tidy3d/components/eme/data/sim_data.py

```diff
@@ -0,0 +1,437 @@
+"""EME simulation data"""
+from __future__ import annotations
+
+from typing import Tuple, Union
+
+import pydantic.v1 as pd
+import numpy as np
+import xarray as xr
+
+from ...base_sim.data.sim_data import AbstractSimulationData
+from ..simulation import EMESimulation
+from .monitor_data import EMEMonitorDataType
+from .dataset import EMESMatrixDataset
+
+from ....log import log
+from ....exceptions import Tidy3dKeyError
+from ...types import Ax, Axis, FieldVal, PlotScale, ColormapType
+from ...viz import equal_aspect, add_ax_if_none
+from ....exceptions import DataError
+from ...data.monitor_data import AbstractFieldData
+
+
+class EMESimulationData(AbstractSimulationData):
+    """Data associated with an EME simulation."""
+
+    simulation: EMESimulation = pd.Field(
+        ..., title="EME simulation", description="EME simulation associated with this data."
+    )
+
+    data: Tuple[EMEMonitorDataType, ...] = pd.Field(
+        ...,
+        title="Monitor Data",
+        description="List of EME monitor data "
+        "associated with the monitors of the original :class:`.EMESimulation`.",
+    )
+
+    smatrix: EMESMatrixDataset = pd.Field(
+        ..., title="S Matrix", description="Scattering matrix of the EME simulation."
+    )
+
+    @staticmethod
+    def apply_phase(data: Union[xr.DataArray, xr.Dataset], phase: float = 0.0) -> xr.DataArray:
+        """Apply a phase to xarray data."""
+        if phase != 0.0:
+            if np.any(np.iscomplex(data.values)):
+                data *= np.exp(1j * phase)
+            else:
+                log.warning(
+                    f"Non-zero phase of {phase} specified but the data being plotted is "
+                    "real-valued. The phase will be ignored in the plot."
+                )
+        return data
+
+    def plot_field(
```

Comment:
Unfortunately, I had to copy this from `components/sim_data.py`. Would it make sense to move some of this field plotting functionality into base_sim?

Reply:
I think what would rather make sense is to have some intermediate level abstract class from which both EME and FDTD simulations inherit (and probably mode in the future, as `ModeSolver` has a `plot_field` method too), but not Heat and e.g. charge in the future. It is either that or decoupling the plot function completely in 3.0, something like `plot_field(sim_data, ...)` instead of `sim_data.plot_field(...)`. But for the time being an `AbstractElectromagneticSimulationData` with a `plot_field` method that both `Simulation` and `EMESimultion` inherit maybe makes most sense?

Reply:
@tylerflex @daquintero

Reply:
that sounds reasonable to me

Reply:
Interestingly, this is the kind of problem we're trying to fix for 3.0 :) Now that we have all these types of simulations, it gets really annoying to have to use inheritance to specify how all of these methods work.

I think the approach momchil suggested probably makes some sense for now.

However, how I'm thinking about it for 3.0 is that the simulation has some properties or methods that simply separate the optical parts out, and then we have some `plot_optical` or `plot_thermal` etc. that filter by these properties and plot them in a generic way.

Related, but I'm thinking that the Simulation and SimulationData might want to be unified in a single object, so `SimulationData.plot_field(mnt_name)` would just work as is now and just continue to error if the `mnt_name` was not a `FieldMonitor`.

Reply:
sorry let me clarify, I dont mean that in 3.0 Simualtion gets unified with SimulationData. I mean that maybe we want to throw EME, Heat, FDTD, etc all into a single Simulation object (and same with their data) basically not distinguishing between the solvers besides the various monitors and sources that are contained within.